api/api-mailhost-v1/src/main/java/com/thoughtworks/go/apiv1/mailhost/representers/MailhostRepresenter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.mailhost.representers;
+
+import com.thoughtworks.go.api.base.OutputWriter;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.config.MailHost;
+import com.thoughtworks.go.spark.Routes;
+
+public class MailhostRepresenter {
+    public static void toJSON(OutputWriter writer, MailHost mailhost) {
+        writer.addLinks(linksWriter -> linksWriter
+                .addLink("self", Routes.MailHost.SELF)
+                .addAbsoluteLink("doc", Routes.MailHost.DOC))
+                .add("hostname", mailhost.getHostName())
+                .add("port", mailhost.getPort())
+                .add("username", mailhost.getUserName())
+                .add("encrypted_password", mailhost.getEncryptedPassword())
+                .add("tls", mailhost.getTls())
+                .add("from", mailhost.getFrom())
+                .add("admin", mailhost.getAdminMail());
+    }
+
+    public static MailHost fromJSON(JsonReader jsonReader) {
+        String hostname = jsonReader.getString("hostname");
+        int port = jsonReader.getInt("port");
```

Comment:
Should be optional int, and throw a validation error from the modal. Not specifying a port will otherwise cause a 422 server error.

Reply:
s/modal/model/

Reply:
@ketan -- Hmm. isn't that applicable for all the APIs?
We perform `get<TYPE>` on mandatory fields, to fail the API when the required fields are not specified.

Server returning 422 in case of not specifying required fields was intentional.